### PR TITLE
TDR-2255 - delete closure metadata

### PR DESCRIPTION
**Pull request by vimleshtna**
If a user clicks the delete metadata button on the Review closure metadata page then show the delete closure metadata confirmation page.
If the user confirms then delete additional closure metadata and redirect to the file navigation page

### CODE REVIEW

**Review by techncl — COMMENTED** [review]
Looks good. Just a few comments.

File: app/controllers/DeleteAdditionalMetadataController.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import graphql.codegen.types.FileFilters
+import org.pac4j.play.scala.SecurityComponents
+import play.api.mvc.{Action, AnyContent, Request}
+import services.{ConsignmentService, CustomMetadataService}
+
+import java.util.UUID
+import javax.inject.Inject
+import scala.concurrent.Future
+
+class DeleteAdditionalMetadataController @Inject()(val consignmentService: ConsignmentService,
```
[the remark this answers is not in the document]
Comment (vimleshtna) [reply app/controllers/DeleteAdditionalMetadataController.scala:15]: We are not sure yet about descriptive metadata. But, yes we may use it for descriptive metadata as well.

File: app/controllers/DeleteAdditionalMetadataController.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            Future(Ok(views.html.standard.confirmDeleteAdditionalMetadata(consignmentId, fileIds, filePaths, request.token.name)))
+          } else {
+            Future.failed(new IllegalStateException(s"Can't find selected files for the consignment $consignmentId"))
+          }
+        }
+        yield response
+      }
+    }
+
+  def deleteAdditionalMetadata(consignmentId: UUID, fileIds: List[UUID]): Action[AnyContent] = standardTypeAction(consignmentId) {
```
[the remark this answers is not in the document]
Comment (vimleshtna) [reply app/controllers/DeleteAdditionalMetadataController.scala:40]: Actually, there is no form as such. It's again a `Get` call to delete the metadata.

File: app/controllers/DeleteAdditionalMetadataController.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  def confirmDeleteAdditionalMetadata(consignmentId: UUID, fileIds: List[UUID]): Action[AnyContent] =
+    standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+      if (fileIds.isEmpty) {
+        Future.failed(new IllegalArgumentException("fileIds are empty"))
+      } else {
+        val filters = Option(FileFilters(None, Option(fileIds), None))
+        for {
+          consignment <- consignmentService.getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, filters)
+          response <- if (consignment.files.nonEmpty) {
+            val filePaths = consignment.files.flatMap(_.fileMetadata).filter(_.name == clientSideOriginalFilepath).map(_.value)
```
[the remark this answers is not in the document]
Comment (vimleshtna) [reply app/controllers/DeleteAdditionalMetadataController.scala:30]: Yeah, It's good but I prefer to use single-line statements. I really don't like `if` in the case block :laughing:

[the remark this answers is not in the document]
Comment (techncl) [reply app/controllers/DeleteAdditionalMetadataController.scala:30]: 😄 fair enough.

File: test/controllers/DeleteAdditionalMetadataControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    wiremockServer.stop()
+  }
+
+  val checkPageForStaticElements = new CheckPageForStaticElements
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  val fileIds: List[UUID] = List(UUID.randomUUID())
+
+  "confirmDeleteAdditionalMetadata" should {
+    "render the delete closure metadata page with an authenticated user" in {
```
Comment (techncl) [inline test/controllers/DeleteAdditionalMetadataControllerSpec.scala:40]: Could you also add a test for an unauthenticated user?

**Review by techncl — COMMENTED** [review]
Looks good, just one comment that you missed from before.

File: test/controllers/DeleteAdditionalMetadataControllerSpec.scala (hunk 4 above)
Comment (vimleshtna) [reply test/controllers/DeleteAdditionalMetadataControllerSpec.scala:40]: We already have one test for an unauthenticated user `redirect to the login page if the page is accessed by a logged out user`. It says logged out user but it is the same.

**Review by techncl — APPROVED** [review]

**Pull request merged**